plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/GeoFunctions.java

```diff
@@ -1188,6 +1189,25 @@ public static Double stDistance(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlTyp
         return leftGeometry.isEmpty() || rightGeometry.isEmpty() ? null : leftGeometry.distance(rightGeometry);
     }
 
+    @SqlNullable
+    @Description("Return the closest points on the two geometries")
+    @ScalarFunction("geometry_nearest_points")
+    @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
```

Comment:
Since the function always returns two points, it would be more natural to return a `row(geometry, geometry)` (with each element being a point) instead of an array containing two elements.

Reply:
@martint  Good point. But it’s a port of the existing functionality and I tried to keep the original code of the author as much as it possible